csharp/src/Apache.Arrow.Flight/Client/FlightClient.cs

```diff
@@ -98,11 +98,25 @@ public AsyncUnaryCall<FlightInfo> GetInfo(FlightDescriptor flightDescriptor, Met
                 flightInfoResult.Dispose);
         }
 
+        [System.Obsolete("Use an async overload that takes a Schema")]
```

Comment:
I'm not sure about whether these overloads should be obsolete, there might be valid use cases where the schema isn't immediately known? But that isn't supported by the Python or C++ implementations.

Reply:
Is there a specific reason to discourage its use?

Reply:
Only to better handle when zero batches are sent. I'm on the fence about this so am happy to remove the `Obsolete` if you think it's too disruptive.

Reply:
I guess I don't have a strong feeling either way.